[docs] Batch13 IP functions (#16947)
apache/druid: docs/querying/sql-functions.md

## CONTAINS_STRING

> Returns true if `str` is a substring of `expr`, case-sensitive. Otherwise returns false.

Reviewer: ```suggestion
Returns true if `str` is a substring of `expr`, case-sensitive. Otherwise, returns false.
```

## ICONTAINS_STRING

> Returns true if `str` is a substring of `expr`, case-insensitive. Otherwise returns false.

Reviewer: ```suggestion
Returns true if `str` is a substring of `expr`, case-insensitive. Otherwise, returns false.
```

## IPV4_MATCH

Returns true if the IPv4 `address` belongs to the `subnet` literal, otherwise returns false.

* **Syntax:** `IPV4_MATCH(address, subnet)`
* **Function type:** Scalar, IP address

<details><summary>Example</summary>

The following example returns true if the IPv4 address in the `forward_for` column from the `kttm` datasource belongs to the subnet `181.13.41.0/24`.

```sql
SELECT 
  "forwarded_for" AS "ipv4_address",
  IPV4_MATCH("forwarded_for", '181.13.41.0/24') AS "belongs_in_subnet"
FROM "kttm"
LIMIT 2
```

Returns the following:

| `ipv4_address` | `belongs_in_subnet`|
| -- | -- | 
| `181.13.41.82` | `true`|
| `177.242.100.0` | `false`|

</details>

[Learn more](sql-scalar.md#ip-address-functions)

## IPV4_PARSE

> Parses `address` into an IPv4 address stored as an integer.

Reviewer 2: this description seems wrong to me based upon the example. I would say:
> Parses an  IPv4 `address` as an integer.

Please verify.

* **Syntax:** `IPV4_PARSE(address)`
* **Function type:** Scalar, IP address

<details><summary>Example</summary>

> The following example returns the IPv4 address `5.5.5.5` represented as an integer.

Reviewer 2: ```suggestion
The following example returns an integer that represents the IPv4 address `5.5.5.5`.
```
avoid passive.

```sql
SELECT 
  '5.5.5.5' AS "ipv4_address",
  IPV4_PARSE('5.5.5.5') AS "integer"
```

Returns the following:

| `ipv4_address` | `integer` |
| -- | -- |
| `5.5.5.5` | `84215045` |

</details>

[Learn more](sql-scalar.md#ip-address-functions)

## IPV4_STRINGIFY

> Converts `address` into an IPv4 address in dot-decimal notation.

Reviewer 2: ```suggestion
Converts an IPv4 `address` in integer notation into dot-decimal notation.
```

* **Syntax:** `IPV4_STRINGIFY(address)`
* **Function type:** Scalar, IP address

<details><summary>Example</summary>

> The following example returns the integer `84215045` in its IPv4 dot-decimal notation.

Reviewer 2: ```suggestion
The following example returns the integer `84215045` in IPv4 dot-decimal notation.
```

## IPV4_MATCH

Returns true if the IPv4 `address` belongs to the `subnet` literal, otherwise returns false.

* **Syntax:** `IPV4_MATCH(address, subnet)`
* **Function type:** Scalar, IP address

<details><summary>Example</summary>

> The following example returns true if the IP address in the `forward_for` column from the `kttm` datasource belongs to the subnet `181.13.41.0/24`.

Reviewer: ```suggestion
The following example returns true if the IPv4 address in the `forward_for` column from the `kttm` datasource belongs to the subnet `181.13.41.0/24`.
```

```sql
SELECT 

>   "forwarded_for" as "ipv4_address",

Reviewer 3: ```suggestion
  "forwarded_for" AS "ipv4_address",
```

| `177.242.100.0` | `false`|

</details>

[Learn more](sql-scalar.md#ip-address-functions)

## IPV4_PARSE

Parses `address` into an IPv4 address stored as an integer.

* **Syntax:** `IPV4_PARSE(address)`
* **Function type:** Scalar, IP address

<details><summary>Example</summary>

The following example returns the IPv4 address `5.5.5.5` represented as an integer.

```sql
SELECT 
  '5.5.5.5' AS "ipv4_address",
  IPV4_PARSE('5.5.5.5') AS "integer"
```

Returns the following:

| `ipv4_address` | `integer` |
| -- | -- |
| `5.5.5.5` | `84215045` |

</details>

[Learn more](sql-scalar.md#ip-address-functions)

## IPV4_STRINGIFY

Converts `address` into an IPv4 address in dot-decimal notation.

* **Syntax:** `IPV4_STRINGIFY(address)`
* **Function type:** Scalar, IP address

<details><summary>Example</summary>

The following example returns the integer `84215045` in its IPv4 dot-decimal notation.

```sql
SELECT 
  '84215045' AS "integer",
  IPV4_STRINGIFY(84215045) AS "dot_decimal_notation"
```

Returns the following:

| `integer` | `dot_decimal_notation` |
| -- | -- |
| `84215045` | `5.5.5.5` |

</details>

[Learn more](sql-scalar.md#ip-address-functions)

## IPV6_MATCH

> Returns true if the IPv6 `address` belongs to the `subnet` literal, else false.

Reviewer: ```suggestion
Returns true if the IPv6 `address` belongs to the `subnet` literal. Otherwise, returns false.
```

* **Syntax:** `IPV6_MATCH(address, subnet)`
* **Function type:** Scalar, IP address

<details><summary>Example</summary>

The following example returns true because `75e9:efa4:29c6:85f6::232c` is in the subnet of `75e9:efa4:29c6:85f6::/64`.

```sql
SELECT 
  '75e9:efa4:29c6:85f6::232c' AS "ipv6_address",
  IPV6_MATCH('75e9:efa4:29c6:85f6::232c', '75e9:efa4:29c6:85f6::/64') AS "belongs_in_subnet" 
```

Returns the following: 

> | `ipv4_address` | `belongs_in_subnet` | 

Reviewer: ```suggestion
| `ipv6_address` | `belongs_in_subnet` | 
```
